### PR TITLE
Fix all paginators from the Applications Tab

### PR DESCRIPTION
**Pull request by afiune**
### :nut_and_bolt: Description
We had a number of issues with our paginators, this change is fixing them all so that the UI and Backend work as expected.

### :+1: Definition of Done
You can use all Paginators in the Applications Tab. That includes when using the health filters! 🎉 

### :athletic_shoe: Demo Script / Repro Steps
Start the studio and build the applications-service. Then build and start the automate-ui in your preferred way. Finally, start all services.
```
$ hab studio enter
[1][default:/src:0]# export DEVPROXY_URL=localhost
[2][default:/src:0]# build components/automate-ui-devproxy
[3][default:/src:0]# build components/applications-service 
[4][default:/src:0]# start_automate_ui_background
[5][default:/src:0]# start_all_services
```

Populate the database (by running the helper method `applications_populate_database`) and navigate in your browser to https://a2-dev.test/applications and use all paginators! 😸 

### :chains: Related Resources
https://chefio.atlassian.net/browse/A2-769

### :white_check_mark: Checklist

- [x] Necessary tests added/updated?
- [x] Necessary docs added/updated?
- [x] Code actually executed?
- [x] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/applications-service/integration_test/service_groups_test.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func TestServiceGroupsBasic(t *testing.T) {
 	assert.Equal(t, expected, response)
 }
 
-func TestServiceGroupsHealthOneOk(t *testing.T) {
+func TestGetServiceGroupsOneOk(t *testing.T) {
```
Comment (afiune) [inline components/applications-service/integration_test/service_groups_test.go]: All these test names where confusing since we are testing the rpc function `GetServicweGroups` not the `ServiceGroupsHealthCounts`.

File: components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.scss [hunk 2]
```diff
@@ -10,9 +10,8 @@ ul {
 }
 
 app-page-picker {
-  margin-left: auto;
-  margin-right: auto;
-  width: 225px;
+  padding-left: 35px;
+  padding-bottom: 35px;
```
Comment (afiune) [inline components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.scss]: This is fixing the margins in the page-picker inside the services-bar, before it was not well aligned.

File: components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.ts [hunk 3]
```diff
@@ -38,34 +43,40 @@ export class ServicesSidebarComponent implements OnInit {
...
 
-    this.servicesHealthSummary$ = this.store.select(createSelector(serviceGroupState,
+    this.svcHealthSummary$ = this.store.select(createSelector(serviceGroupState,
       (state) => state.servicesHealthSummary));
+    this.svcHealthSummary$.subscribe((servicesHealthSummary) => {
+      this.servicesHealthSummary = servicesHealthSummary;
+      this.totalServices = getOr(0, this.selectedHealth, this.servicesHealthSummary);
+    });
 
-    this.currentPage = 1;
```
Comment (afiune) [inline components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.ts]: This page reset to default is now happening inside the `openServicesSidebar()` functions on the service-groups.component.ts. (https://github.com/chef/automate/pull/218/files#diff-f10f08039008f57ac8b564e592765087R154)

Why the change? - Because the OnInit function happens only once when the component is created in the browser, and not every time the bar is visible or invisible.

File: components/applications-service/integration_test/global_test.go [hunk 4]
```diff
@@ -76,3 +78,73 @@ func TestMain(m *testing.M) {
 	// call with result of m.Run()
 	os.Exit(exitCode)
 }
+
+// assertServiceGroupsEqual verifies that the two provided ServiceGroup Array are equal
+func assertServiceGroupsEqual(t *testing.T, expected, actual *applications.ServiceGroups) {
```
Comment (danielsdeleo) [inline components/applications-service/integration_test/global_test.go]: Oh man, this is a super nice way to make failing tests easier to understand 💯
